src/control/Control.cpp

```diff
@@ -2485,8 +2101,8 @@ bool Control::loadMetadataCallback(MetadataCallbackData* data)
 
 void Control::loadMetadata(MetadataEntry md)
 {
-	MetadataCallbackData* data = new MetadataCallbackData();
-	data->md = md;
+	auto* data = new MetadataCallbackData();
+	data->md = std::move(md);
```

Comment:
MetadataEntry does not have a move constructor, therefore I think this brings nothing?

Reply:
It does. I don't have the source in front of me, but even the copy constructor isn't defined explicit it is defined implicit, because a child is move constructible. E.g. a std::string ord std::vector.

Reply:
Ok. Tested and confirmed, it's faster.

```
	MetadataEntry e1;
	e1.path = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
	
	for(long i = 0; i < 10000000; i++) {
		MetadataEntry e2 = e1;
	}
```
and the second example with
```
		MetadataEntry e2 = std::move(e1);
```

```
time ./t1

real	0m0.285s
user	0m0.285s
sys	0m0.001s
```

And with move
```
time ./t2

real	0m0.131s
user	0m0.131s
sys	0m0.001s
```

Therefore this really brings something, even if it's really less...

Reply:
The different between copy and move is basically, copy makes a deep copy, move a flat copy + invalidating the old Object (set all ptrs to nullptr; ). So every Object owning pointers is faster to move than to copy, since we don't need to allocate and deallocate the old and new ptr. It also is exception safe.